src/Assembly.js

```diff
@@ -274,11 +284,44 @@ class Assembly extends React.Component {
   }
 
   reportMedication() {
-    this.registration.create("medication_reports", { timestamp: `${this.survey_date}T${this.survey_medication_time}:00.000` }, this.registration.information.uuid)
+    if (this.survey_tookMedication != null) {
+      this.registration.create(
+        "medication_reports", 
+        { 
+          timestamp: `${this.survey_date}T${this.survey_medication_time}:00.000`,
+          took_medication: this.survey_tookMedication,
+          not_taking_medication_reason: this.survey_notTakingMedicationReason 
+        },
+        this.registration.information.uuid
+      )
+    }
+    
+    this.survey_date = null
+    this.survey_medication_time = null
```

Comment:
We need to set this back to the original value from the constructor, not `null`:

```suggestion
    this.survey_medication_time = DateTime.local().setLocale(this.locale).toLocaleString(DateTime.TIME_24_SIMPLE)
```